pytensor/tensor/slinalg.py

```diff
@@ -908,6 +908,101 @@ def solve_discrete_are(A, B, Q, R, enforce_Q_symmetric=False) -> TensorVariable:
     )
 
 
+def largest_common_dtype(tensors: typing.Sequence[TensorVariable]) -> np.dtype:
+    return ft.reduce(lambda l, r: np.promote_types(l, r), [x.dtype for x in tensors])
+
+
+def block_diag_grad(inputs, gout):
+    shapes = pt.stack([i.shape for i in inputs])
+    index_end = shapes.cumsum(0)
+    index_begin = index_end - shapes
+    slices = [
+        ptb.ix_(
+            pt.arange(index_begin[i, 0], index_end[i, 0]),
+            pt.arange(index_begin[i, 1], index_end[i, 1]),
+        )
+        for i in range(len(inputs))
+    ]
+    return [gout[0][slc] for slc in slices]
+
+
+class BaseBlockDiagonal(Op):
+    def grad(self, inputs, gout):
+        return block_diag_grad(inputs, gout)
+
+    def infer_shape(self, fgraph, nodes, shapes):
+        first, second = zip(*shapes)
+        return [(pt.add(*first), pt.add(*second))]
+
+
+class BlockDiagonalMatrix(BaseBlockDiagonal):
+    def make_node(self, *matrices, name=None):
```

Comment:
Never saw name in make_node, can it be reached from `op.__call__`? That would actually be a good place to allow a name kwarg, so it would work with all Ops!

Reply:
It breaks `Blockwise` but works otherwise

Reply:
What breaks Blockwise?

Reply:
Having kwargs in `make_node`

Reply:
make_node shouldn't have kwargs I think, but Op.__call__ can and name would be a neat one to add?

Reply:
Oh interesting! I'll have a look at that. I really like having names for the stack traces and dprints, so i'd like it if every Op had it.

Reply:
Me too, it's also awkward to need two lines to give a name